### PR TITLE
Connect bundle details page and sidebar with REST API

### PR DESCRIPTION
**Pull request by kashizui**
Depends on codalab/codalab-cli#425

### CODE REVIEW

File: codalab/apps/web/static/js/bundle/bundle_interface.jsx [hunk 1]
```diff
@@ -33,56 +31,85 @@ var Bundle = React.createClass({
...
+            if (new_metadata[key].constructor === Array) {
+                // Convert string comma-separated list to Array
+                val = val.split(',');
+            }
+            if (val != new_metadata[key]) {
+                new_metadata[key] = val;
+            }
         });
 
         console.log('------ save the bundle here ------');
```
Comment (percyliang) [inline codalab/apps/web/static/js/bundle/bundle_interface.jsx]: Can remove these debugging statements

File: codalab/apps/web/static/js/bundle/bundle_interface.jsx [hunk 2]
```diff
@@ -33,56 +31,85 @@ var Bundle = React.createClass({
...
+            data: {
+                include: 'users,bundle-permissions'
+            },
             dataType: 'json',
             cache: false,
             success: function(data) {
-                if(this.isMounted()){
-                    this.setState(data);
+                // Use ephemeral JsonApiDataStore to wire up relationships
+                var store = new JsonApiDataStore();
```
Comment (percyliang) [inline codalab/apps/web/static/js/bundle/bundle_interface.jsx]: can put this line inside the if

File: codalab/apps/web/static/js/bundle/bundle_interface.jsx [hunk 3]
```diff
@@ -236,7 +269,7 @@ var Bundle = React.createClass({
 
         /// ------------------------------------------------------------------
         var edit = ''
-        if(this.state.edit_permission){
+        if(this.state.permission >= 2){
```
Comment (percyliang) [inline codalab/apps/web/static/js/bundle/bundle_interface.jsx]: define 2 as EDIT_PERMISSION like we do in Python

**Comment by percyliang** [comment]
LGTM modulo comments

**Comment by kashizui** [comment]
@percyliang this PR was waiting on the completion of the REST API and another issue (now closed) to merge the bundle details and sidebar code. It's pretty out of date now but can be revamped and prepared for merging in the future.

**Pull request closed without merging**